overlord/snapstate/snapmgr.go

```diff
@@ -99,9 +104,46 @@ func (m *SnapManager) doInstallSnap(t *state.Task, _ *tomb.Tomb) error {
 	}
 	t.State().Unlock()
 
+	// see if it is a local snap, if we, nothing to download
+	if fi, err := os.Stat(inst.Name); err == nil && fi.Mode().IsRegular() {
+		inst.DownloadedSnap = inst.Name
+		inst.Developer = snappy.SideloadedDeveloper
+	} else {
+		pb := &TaskProgressAdapter{task: t}
+		downloadedSnapFile, developer, err := m.backend.Download(inst.Name, inst.Channel, pb)
+		if err != nil {
+			return err
+		}
+		inst.DownloadedSnap = downloadedSnapFile
+		inst.Developer = developer
+	}
+
+	// update instState for the next task
+	t.State().Lock()
+	t.Set("install-state", inst)
+	t.State().Unlock()
+
+	return nil
+}
+
+func (m *SnapManager) doInstallLocalSnap(t *state.Task, _ *tomb.Tomb) error {
+	var inst installState
+	var dlTaskID string
+
+	t.State().Lock()
+	if err := t.Get("download-task-id", &dlTaskID); err != nil {
```

Comment:
Or maybe we might already have reasons to diverge them? It feels like this should be next to the rest of the data we used for handling this task. If we need to diverge the install-state at some point, it'll probably be because we changed the semantics enough to guarantee drawing a line there.